test/services/PensionSessionServiceSpec.scala

```diff
@@ -128,4 +130,50 @@ class PensionSessionServiceSpec extends UnitTest with MockPensionUserDataReposit
       response shouldBe a[DatabaseError]
     }
   }
+
+  "mergePriorDataToSession" should {
+    val noPriorData = IncomeTaxUserData(None)
+
+    "return a successful result if the model required updating (session != prior)" in {
+      super[MockIncomeTaxUserDataConnector].mockFind(nino, taxYear, noPriorData)
+      super[MockPensionUserDataRepository].mockFind(taxYear, user, Right(None))
+      mockCreateOrUpdate(Right(()))
+
+      val response = service.mergePriorDataToSession(taxYear, user, (_, _, _) => Ok)
+
+      assert(status(response) === OK)
+    }
+
+    "return a successful if not update required (session == prior)" in {
+      val sessionWithEmptyModel = aPensionsUserData.copy(pensions = PensionsCYAModel.emptyModels)
+      super[MockIncomeTaxUserDataConnector].mockFind(nino, taxYear, noPriorData)
+      super[MockPensionUserDataRepository].mockFind(taxYear, user, Right(Some(sessionWithEmptyModel)))
+      mockCreateOrUpdate(Right(()))
+
+      val response = service.mergePriorDataToSession(taxYear, user, (_, _, _) => Ok)
+
+      assert(status(response) === OK)
+    }
+
+    "return an internal server error if data not found" in {
+      super[MockIncomeTaxUserDataConnector].mockFind(nino, taxYear, noPriorData)
+      super[MockPensionUserDataRepository].mockFind(taxYear, user, Left(DataNotFound))
+      mockCreateOrUpdate(Right(()))
```

Comment:
I don't think we need this mock call here.

Reply:
yes, removed